susemanager/bin/pg-migrate-10-to-13.sh

```diff
@@ -0,0 +1,129 @@
+        echo
+        echo "`date +"%H:%M:%S"`   If you have a backup of the database, run \"$0 fast\""
+        echo
+        exit 1
+    else
+        echo "ok."
+    fi
+fi
+
+echo "`date +"%H:%M:%S"`   Shut down spacewalk services..."
+spacewalk-service stop
+systemctl stop postgresql
+
+echo "`date +"%H:%M:%S"`   Checking postgresql version..."
+rpm -q postgresql$NEW_VERSION-server > /dev/null 2>&1
+if [ $? -eq 0 ]; then
+    echo "`date +"%H:%M:%S"`   postgresql $NEW_VERSION is already installed. Good."
+else
+    echo "`date +"%H:%M:%S"`   Installing postgresql $NEW_VERSION..."
+    zypper --non-interactive in postgresql$NEW_VERSION postgresql$NEW_VERSION-contrib postgresql$NEW_VERSION-server
+
+    if [ ! $? -eq 0 ]; then
+        echo "`date +"%H:%M:%S"`   Installation of postgresql $NEW_VERSION failed!"
+        exit 1
+    fi
+fi
+
+echo "`date +"%H:%M:%S"`   Ensure postgresql $NEW_VERSION is being used as default..."
+/usr/sbin/update-alternatives --set postgresql /usr/lib/postgresql$NEW_VERSION
+if [ $? -eq 0 ]; then
+    echo "`date +"%H:%M:%S"`   Successfully switched to new postgresql version $NEW_VERSION."
+else
+    echo "`date +"%H:%M:%S"`   Could not switch to new postgresql version $NEW_VERSION!"
+    exit 1
+fi
+
+echo "`date +"%H:%M:%S"`   Create new database directory..."
+mv /var/lib/pgsql/data /var/lib/pgsql/data-pg$OLD_VERSION
+mkdir /var/lib/pgsql/data
+chown postgres:postgres /var/lib/pgsql/data
+
+echo "`date +"%H:%M:%S"`   Initialize new postgresql $NEW_VERSION database..."
+. /etc/sysconfig/postgresql
+if [ -z $POSTGRES_LANG ]; then
+    POSTGRES_LANG="en_US.UTF-8"
+fi
+su -s /bin/bash - postgres -c "initdb -D /var/lib/pgsql/data --locale=$POSTGRES_LANG"
+if [ $? -eq 0 ]; then
+    echo "`date +"%H:%M:%S"`   Successfully initialized new postgresql $NEW_VERSION database."
+else
+    echo "`date +"%H:%M:%S"`   Initialization of new postgresql $NEW_VERSION database failed!"
+    echo "`date +"%H:%M:%S"`   Trying to restore previous state..."
+    mv /var/lib/pgsql/data /var/lib/pgsql/data-new-failed
+    mv /var/lib/pgsql/data-pg10 /var/lib/pgsql/data
+    /usr/sbin/update-alternatives --set postgresql /usr/lib/postgresql10
+    exit 1
+fi
+
+echo "`date +"%H:%M:%S"`   Upgrade database to new version postgresql $NEW_VERSION..."
+su -s /bin/bash - postgres -c "pg_upgrade --old-bindir=/usr/lib/postgresql$OLD_VERSION/bin --new-bindir=/usr/lib/postgresql$NEW_VERSION/bin --old-datadir=/var/lib/pgsql/data-pg$OLD_VERSION --new-datadir=/var/lib/pgsql/data $FAST_UPGRADE"
+if [ $? -eq 0 ]; then
+    echo "`date +"%H:%M:%S"`   Successfully upgraded database to postgresql $NEW_VERSION."
+else
+    echo "`date +"%H:%M:%S"`   Upgrading database to version $NEW_VERSION failed!"
+    echo "`date +"%H:%M:%S"`   Trying to restore previous state..."
+    mv /var/lib/pgsql/data /var/lib/pgsql/data-new-failed
+    mv /var/lib/pgsql/data-pg$OLD_VERSION /var/lib/pgsql/data
```

Comment:
I strongly recommend you always use` ${NAME}. It improves readability, and also is safer as it works regardless of what's around the variable :-)

Reply:
yeah, I thought about that, but then I have to change that to the rest of the script to be consistent, and I thought it was too much refactor at this moment